internal/cmd/argus/instance/update/update_test.go

```diff
@@ -0,0 +1,451 @@
+					}
+					t.Fatalf("setting flag --%s=%s: %v", flag, value, err)
+				}
+			}
+
+			err = cmd.ValidateArgs(tt.argValues)
+			if err != nil {
+				if !tt.isValid {
+					return
+				}
+				t.Fatalf("error validating args: %v", err)
+			}
+
+			err = cmd.ValidateRequiredFlags()
+			if err != nil {
+				if !tt.isValid {
+					return
+				}
+				t.Fatalf("error validating flags: %v", err)
+			}
+
+			model, err := parseInput(cmd, tt.argValues)
+			if err != nil {
+				if !tt.isValid {
+					return
+				}
+				t.Fatalf("error parsing flags: %v", err)
+			}
+
+			if !tt.isValid {
+				t.Fatalf("did not fail on invalid input")
+			}
+			diff := cmp.Diff(model, tt.expectedModel)
+			if diff != "" {
+				t.Fatalf("Data does not match: %s", diff)
+			}
+		})
+	}
+}
+
+func TestBuildRequest(t *testing.T) {
+	tests := []struct {
+		description         string
+		model               *inputModel
+		expectedRequest     argus.ApiUpdateInstanceRequest
+		getPlansFails       bool
+		getPlansResponse    *argus.PlansResponse
+		getInstanceFails    bool
+		getInstanceResponse *argus.GetInstanceResponse
+		isValid             bool
+	}{
+		{
+			description:         "base",
+			model:               fixtureInputModel(),
+			expectedRequest:     fixtureRequest(),
+			getPlansResponse:    fixturePlansResponse(),
+			getInstanceResponse: fixtureGetInstanceResponse(),
+			isValid:             true,
+		},
+		{
+			description: "use plan name",
+			model: fixtureInputModel(
+				func(model *inputModel) {
+					model.PlanId = nil
+					model.PlanName = "example-plan-name"
+				},
+			),
+			expectedRequest:     fixtureRequest(),
+			getPlansResponse:    fixturePlansResponse(),
+			getInstanceResponse: fixtureGetInstanceResponse(),
+			isValid:             true,
+		},
+		{
+			description: "get plans fails",
+			model: fixtureInputModel(
+				func(model *inputModel) {
+					model.PlanId = nil
+					model.PlanName = "example-plan-name"
+				},
+			),
+			getPlansFails: true,
+			isValid:       false,
+		},
+		{
+			description: "plan name not found",
+			model: fixtureInputModel(
+				func(model *inputModel) {
+					model.PlanId = nil
+					model.PlanName = "non-existent-plan"
+				},
+			),
+			getPlansResponse:    fixturePlansResponse(),
+			getInstanceResponse: fixtureGetInstanceResponse(),
+			isValid:             false,
+		},
+		{
+			description: "plan id not found",
+			model: fixtureInputModel(
+				func(model *inputModel) {
+					model.PlanId = utils.Ptr(uuid.NewString())
+				},
+			),
+			getPlansResponse:    fixturePlansResponse(),
+			getInstanceResponse: fixtureGetInstanceResponse(),
+			isValid:             false,
+		},
+		{
+			description: "plan id, no instance name",
+			model: fixtureInputModel(
+				func(model *inputModel) {
+					model.InstanceName = nil
+				},
+			),
+			getPlansResponse:    fixturePlansResponse(),
+			getInstanceResponse: fixtureGetInstanceResponse(),
+			expectedRequest:     fixtureRequest(),
+			isValid:             true,
+		},
```

Comment:
In these tests we should test updating the plan to a new plan, to test that the built request contains the `NewPlanId` and not the current `TestPlanId`

Currently it is only testing that the request contains *a* plan ID (which should always be true)